File: 538. Convert BST to Greater Tree.ts
function convertBST(root: TreeNode | null): TreeNode | null {
  let sum = 0;

  const traverse = (node: TreeNode | null): TreeNode | null => {
    if (!node) return;

    traverse(node.right);
    node.val += sum;
    sum = node.val;
    traverse(node.left);
  };

  traverse(root);
  return root;
}

File: Leetcode/538. Convert BST to Greater Tree.py
class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        def traverse(node: TreeNode):
            if not node:
                return
            nonlocal total
            traverse(node.right)
            total += node.val
            node.val = total
            traverse(node.left)

        total = 0
        traverse(root)
        return root
